Source/Core/Cartesian3.js

```diff
@@ -58,9 +60,7 @@ define([
      */
     Cartesian3.fromSpherical = function(spherical, result) {
         //>>includeStart('debug', pragmas.debug);
-        if (!defined(spherical)) {
-            throw new DeveloperError('spherical is required');
-        }
+        check.type.object(spherical);
```

Comment:
Could be fine, but try something that is math heavily like triangulating a polygon or a microbenchmark with Cartesians.  Removing the developer errors can save 2-3% in release builds, as long as this doesn't jump to, I dunno, 10%, I think we are good.